Utilities/src/test/java/edu/stanford/hivdb/utilities/MutationFileReaderTest.java

```diff
@@ -18,37 +18,65 @@
 
 package edu.stanford.hivdb.utilities;
 
-/*import java.io.InputStream;
+import static org.junit.Assert.assertEquals;
+
+import java.io.FileNotFoundException;
+import java.io.InputStream;
 import java.util.List;
 
 import org.junit.Test;
 
-import edu.stanford.hivdb.utilities.MutationFileReader;
-import edu.stanford.hivdb.mutations.Gene;
+import edu.stanford.hivdb.filetestutils.TestMutationsFiles;
+import edu.stanford.hivdb.filetestutils.TestMutationsFiles.TestMutationsProperties;
+//import edu.stanford.hivdb.drugresistance.TestMutationPatternFiles;
+//import edu.stanford.hivdb.drugresistance.TestMutationPatternFiles.TestMutationPatterns;
+//import edu.stanford.hivdb.drugs.DrugClass;
 import edu.stanford.hivdb.mutations.MutationSet;
-import edu.stanford.hivdb.testutils.TestMutationsFiles;
-import edu.stanford.hivdb.testutils.TestMutationsFiles.TestMutationsProperties;*/
 
 public class MutationFileReaderTest {
 
-	/*@Test
-	public void test() {
-		final InputStream testMutationsInputStream = TestMutationsFiles.getTestMutationsInputStream(TestMutationsProperties.SIMPLE_MUTATIONS_TEST);
-		final List<MutationSet> mutationLists = MutationFileReader.readMutationLists(testMutationsInputStream);
-
-		int numMutLists = mutationLists.size();
-		System.out.println("The file " + TestMutationsProperties.SIMPLE_MUTATIONS_TEST +
-				" contained " + numMutLists + " mutation lists.");
-		for (MutationSet mutList : mutationLists) {
-			System.out.println();
-			for (Gene gene : Gene.values()) {
-				System.out.printf("%s:", gene);
-				MutationSet geneMutList = mutList.getGeneMutations(gene);
-				if (geneMutList.size() > 0) {
-					System.out.println(geneMutList.join());
-				}
-			}
-		}
-	}*/
-
-}
+	@Test
+	public void  testReadMutationLists() {
+		final InputStream mutsStream =
+				TestMutationsFiles.getTestMutationsInputStream(TestMutationsProperties.SIMPLE_MUTATIONS_TEST);
+		final List<MutationSet> mutationLists = MutationFileReader.readMutationLists(mutsStream);
+		final String eMutSet0 = "[M36I, I54V, V82F, M41L, K103N, M184V, T215Y, Q148H]";
+		final String eMutSet5 = "[M36I, V82F, M41L, K103N, M184V, T215SY, Q148H]";
+		final String mutSet0 = mutationLists.get(0).toString();
+		final String mutSet5 = mutationLists.get(5).toString();
+		assertEquals(6, mutationLists.size());
+		assertEquals(eMutSet0, mutSet0);
+		assertEquals(eMutSet5, mutSet5);
+	}
+		
+	@Test
+	public void testReadMutationListsForDrugClass() throws FileNotFoundException {
```

Comment:
Calling `TestMutationPatternFiles.getTestMutationPatternsInputStream()` requires us to add the DrugResistance project to `Utilities/build.gradle` which would introduce a circular inclusion path since DrugResistance compiles Utilities in `DrugResistance/build.gradle`. Hence we should ultimately move `TestMutationPatternFiles` to the TestResources project like we did with `TestMutationFiles`

Reply:
Mutation pattern files are relied on drug resistance interpretation. And as I said they are automatically generated by a script from `DrugResistance/.../tests`. If you want to test a whole bunch of mutation lists I suggest you to create duplications in `TestResources` but don't use the word "Mutation Pattern".